--- bin/install-catalog.in	2001-01-12 16:35:39.000000000 +0100
+++ bin/install-catalog.in.new	2024-03-30 16:06:03.097457324 +0100
@@ -63,23 +63,23 @@
 esac
 
 # Check that the super catalog can be created and changed and deleted
-if [ ! -w /etc/sgml ]
+if [ ! -w $PREFIX/etc/sgml ]
 then
-  echo "`basename $0`: unable to write in /etc/sgml." >&2
+  echo "`basename $0`: unable to write in $PREFIX/etc/sgml." >&2
   exit 2
 fi
 case $SGML_ACTION in
    addition)
-	if [ -e /etc/sgml/catalog -a ! -w /etc/sgml/catalog ]
+	if [ -e $PREFIX/etc/sgml/catalog -a ! -w $PREFIX/etc/sgml/catalog ]
 	then
-	  echo "`basename $0`: can not modify \"/etc/sgml/catalog\"." >&2
+	  echo "`basename $0`: can not modify \"$PREFIX/etc/sgml/catalog\"." >&2
 	  exit 2
 	fi
 	;;
    removal)
-	if [ ! -w /etc/sgml/catalog ]
+	if [ ! -w $PREFIX/etc/sgml/catalog ]
 	then
-	  echo "`basename $0`: can not modify \"/etc/sgml/catalog\"." >&2
+	  echo "`basename $0`: can not modify \"$PREFIX/etc/sgml/catalog\"." >&2
 	  exit 2
 	fi
 	;;
@@ -134,11 +134,11 @@
 	else
 	  echo "$SGML_POINTER $SGML_ORDINARY" >> $SGML_CENTRALIZED
 	fi
-	grep -q $SGML_CENTRALIZED /etc/sgml/catalog 2>/dev/null
+	grep -q $SGML_CENTRALIZED $PREFIX/etc/sgml/catalog 2>/dev/null
 	if [ $? -ne 0 ]
 	then
-	  echo "`basename $0`: addition of $SGML_CENTRALIZED in /etc/sgml/catalog"
-	  echo "$SGML_POINTER $SGML_CENTRALIZED" >> /etc/sgml/catalog
+	  echo "`basename $0`: addition of $SGML_CENTRALIZED in $PREFIX/etc/sgml/catalog"
+	  echo "$SGML_POINTER $SGML_CENTRALIZED" >> $PREFIX/etc/sgml/catalog
 	fi
 	;;
    removal)
@@ -153,9 +153,9 @@
 	if [ ! -s $SGML_CENTRALIZED ]
 	then
 	  rm $SGML_CENTRALIZED
-	  echo "`basename $0`: removal of $SGML_CENTRALIZED from /etc/sgml/catalog"
-          sed -e "\:$SGML_POINTER $SGML_CENTRALIZED:d" < /etc/sgml/catalog > /etc/sgml/catalog.new
-          mv /etc/sgml/catalog.new /etc/sgml/catalog
+	  echo "`basename $0`: removal of $SGML_CENTRALIZED from $PREFIX/etc/sgml/catalog"
+          sed -e "\:$SGML_POINTER $SGML_CENTRALIZED:d" < $PREFIX/etc/sgml/catalog > $PREFIX/etc/sgml/catalog.new
+          mv $PREFIX/etc/sgml/catalog.new $PREFIX/etc/sgml/catalog
 	fi
 	;;
 esac
